### PR TITLE
Improve InternalModel (2x - 3x faster) 

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

File: packages/ember-data/lib/system/model/internal-model.js [hunk 1]
```diff
@@ -2,13 +2,14 @@ import merge from "ember-data/system/merge";
...
+import EmptyObject from "ember-data/system/empty-object";
 
 var Promise = Ember.RSVP.Promise;
 var get = Ember.get;
 var set = Ember.set;
 
-var _extractPivotNameCache = Object.create(null);
-var _splitOnDotCache = Object.create(null);
+var _extractPivotNameCache = new EmberObject();
+var _splitOnDotCache = new EmberObject();
```
Comment (dschmidt) [inline packages/ember-data/lib/system/model/internal-model.js]: Didn't you mean EmptyObject here too?

**Comment by stefanpenner** [comment]
Lol late night patches. Will fix when I get to work

**Comment by fivetanley** [comment]
How do we measure the speed?

**Comment by stefanpenner** [comment]
> How do we measure the speed?

how do you, or how did I?

I just through together a quick one-off manual test.

**Comment by stefanpenner** [comment]
i'll look at the tests shortly, i was an idiot and started moving other stuff around :P

**Comment by stefanpenner** [comment]
i think travis is in a bad state, the UI seems funky telling me this project isn't setup/configured.

Tests appear green locally.

**Comment by stefanpenner** [comment]
@fivetanley @wecc @igorT any ideas whats up with travis and ED ?

screen shot 2015-08-12 at 11 57 37 am

**Comment by stefanpenner** [comment]
looks like a test failure unrelated to this PR, it is present on master as is.

Investigating

**Pull request merged**